day1/es1/LocalButton.qml

```diff
@@ -0,0 +1,33 @@
+import QtQuick 2.0
+
+Rectangle {
+    id: button
+
+    property string stateName: "unchecked"
+    property color colorForTheStateChecked: "lightgray"
+    property color colorForTheStateUnchecked: "lightgray"
+    property string text: qsTr("none")
+
+    color: switchButton.state == "checked"? colorForTheStateChecked: colorForTheStateUnchecked
+    width: parent.width/parent.children.length
+    height: parent.height
+
+
+    Text {
+        anchors.centerIn: parent
+        text: button.text
+    }
+
+    MouseArea {
+        anchors.fill: parent
+        onClicked: {
+            switchButton.state = stateName
```

Comment:
Se ci fai caso, `switchButton` è di un colore particolare (a seconda del tema che hai).
In questo caso, significa che la componente non è in grado di identificare l'ID dal nome `switchButton` all'interno di questo file, e quindi lo sta cercando a livello globale.
In questo caso non è un problema, perchè nel progetto c'è un solo `switchButton`, ma se ne avessimo più d'uno potremmo avere problemi. 
Generalmente ciascuna componente dovrebbe essere auto-contenuta: questo significa che dovrebbe ignorare tutto quello che c'è aldifuori di lei, e lavorare solo con le cose contenute al suo interno.

In questo caso potremmo fare così:

- aggiungere un `signal clicked` sotto le property che hai già definito in alto, così stiamo dicendo che la componente LocalButton può emettere un segnale chiamato `clicked` che gestiremo da fuori con un handler `onClicked`
- al posto di `switchButton.state = stateName` possiamo emettere il segnale appena definito in questo modo: `button.clicked()`
- la gestione del segnale la spostiamo quindi in SwitchButton.qml, dentro ciascun LocalButton

In questo modo riusciamo a evitare di dipendere da un ID globale! :D

Reply:
eliminato utlizzo ID globale